pkg/mdns/controller.go

```diff
@@ -3,11 +3,11 @@ package mdns
 import (
 	"context"
 	"net"
-	"regexp"
 	"strings"
 	"sync"
 
 	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/config/ovn"
```

Comment:
This introduces a dependency between the MDNS and OVN components, which is going to make it harder to make OVN pluggable. We can do this for now, but we need a plan to decouple these again in the future. What can we do to keep them loosely coupled?

Reply:
I'm actually doing some code preparation work in this PR to move ovnk related code in its own files under  `pkg/config/ovn`. Before this PR, the ovnk interfaces and special IP address are hardcoded in mdns controller. For fully decoupling ovnk from mdns controller (assuming mdns will stay in microshift main), we would need an interface between microshift and  plugins to exchange the mdns related information (this is where the binary/script method in pluggable enhancement would help).